deployments/engine/docker-compose/dm_databases_with_tls.yaml

```diff
@@ -0,0 +1,65 @@
+version: '2.3'
+
+services:
+    dm_upstream_mysql:
+        image: mysql:5.7
+        container_name: dm_upstream_mysql
+        command:
+            - --require_secure_transport=ON
+        ports:
+            - "3306:3306"
+        volumes:
+            - ./config/mysql.cnf:/etc/mysql/conf.d/mysql.cnf
+        environment:
+            MYSQL_ROOT_PASSWORD: 123456
+
+    dm_upstream_mysql2:
+        image: mysql:8.0
+        container_name: dm_upstream_mysql2
+        command:
+            - --require_secure_transport=ON
+        ports:
+            - "3307:3306"
+        environment:
+            MYSQL_ROOT_PASSWORD: 123456
+
+    dm_downstream_pd:
+        image: pingcap/pd:nightly
+        ports:
+            - "2379"
+        command:
+            - --name=dm_downstream_pd
+            - --client-urls=http://0.0.0.0:2379
+            - --peer-urls=http://0.0.0.0:2380
+            - --advertise-client-urls=http://dm_downstream_pd:2379
+            - --advertise-peer-urls=http://dm_downstream_pd:2380
+            - --initial-cluster=dm_downstream_pd=http://dm_downstream_pd:2380
+            - --data-dir=/data/dm_downstream_pd
+        restart: on-failure
+
+    dm_downstream_tikv:
+        image: pingcap/tikv:nightly
+        command:
+            - --addr=0.0.0.0:20160
+            - --advertise-addr=dm_downstream_tikv:20160
+            - --pd=dm_downstream_pd:2379
+        depends_on:
+            - "dm_downstream_pd"
+        restart: on-failure
+
+    dm_downstream_tidb:
+        image: pingcap/tidb:latest
+        ports:
+            - "4000:4000"
+            - "10080:10080"
+        volumes:
+            - ./config/tidb_tls.toml:/tidb.toml
+            - /tmp/certs/downstream:/etc/certs
```

Comment:
Better to use `/tmp/tiflow_engine_test` ?

Reply:
it will be adjusted to wrong path

Reply:
The adjust logic will cause many unexpected behaviour, and it's really hard to locate. I don't like it.

Reply:
From my point of view, the adjust logic was a compromise that allowed multiple integration tests to share a single deployment file while still handling log files appropriately. And using '/tmp/tiflow_engine_test' ensures that all configuration files for an integration test are clustered in the same directory, which can greatly reduce the burden of finding test-related files. 
So even though 'adjust_config' has some problems, we should first try to fix it rather than scrap it. Besides, can you list some unexpected behaviors?

Reply:
I wrote `/tmp/tiflow_engine_test/certs/downstream` here, and also align this path in for `generate_cert` and mysql client in`run.sh`. And as the result, mysql client to TiDB report "server does not support TLS". Very confusing.

Please fix it in another PR, this PR has been opened too long.